### PR TITLE
make dashboard version card a functional component

### PR DESCRIPTION
**Pull request by rianfowler**
#### What this PR does / why we need it:
- refactors DashboardVersionCard into a functional component so we can use react-query hooks in it 

#### Which issue(s) this PR fixes:

Fixes #

https://app.shortcut.com/replicated/story/50084/fix-multi-app-ux

#### Does this PR introduce a user-facing change?

none- refactor 

#### Does this PR require documentation?
none

### CODE REVIEW

File: web/src/components/apps/DashboardVersionCard.tsx [hunk 1]
```diff
@@ -732,134 +833,38 @@ class DashboardVersionCard extends React.Component<Props, State> {
...
 
-  showReleaseNotes = (releaseNotes: string) => {
-    this.setState({
-      showReleaseNotes: true,
-      releaseNotes: releaseNotes,
-    });
-  };
-
-  hideReleaseNotes = () => {
-    this.setState({
```
Comment (rianfowler) [inline web/src/components/apps/DashboardVersionCard.tsx:846]: as far as i can tell this `showReleaseNotes` is never called so removing for now.

File: web/src/components/apps/DashboardVersionCard.tsx [hunk 2]
```diff
@@ -1485,456 +1487,443 @@ class DashboardVersionCard extends React.Component<Props, State> {
...
+            </button>
+          </div>
+        </Modal>
+      )}
+    </div>
+  );
+};
 
-// eslint-disable-next-line
-export default withRouter(DashboardVersionCard) as any;
```
Comment (rianfowler) [inline web/src/components/apps/DashboardVersionCard.tsx:1940]: added react router hooks so we can get rid of this

**Review by rianfowler — COMMENTED** [review]
noted the main changes. the rest of the changes are just putting functions into variables, refactoring out `this.` and moving functions around into a valid order

File: web/src/components/apps/Dashboard.tsx [hunk 3]
```diff
@@ -48,9 +48,14 @@ type Props = {
...
-  makeCurrentVersion: (version: Version) => void;
+  makeCurrentVersion: (
+    slug: string,
+    versionToDeploy: Version,
+    isSkipPreflights: boolean,
+    continueWithFailedPreflights: boolean
+  ) => void;
   ping: (clusterId?: string) => void;
-  redeployVersion: (version: Version) => void;
+  redeployVersion: (slug: string, version: Version | null) => void;
```
Comment (rianfowler) [inline web/src/components/apps/Dashboard.tsx:58]: these are just passed through to the version card. we'll need to refactor out this prop drilling eventually

**Review by miaawong — APPROVED** [review]

**Pull request merged**